plugins/com.google.cloud.tools.eclipse.sdk.ui/src/com/google/cloud/tools/eclipse/sdk/ui/preferences/CloudSdkPreferenceArea.java

```diff
@@ -302,13 +302,13 @@ private boolean validateSdk(Path location) {
       // accept a seemingly invalid location in case the SDK organization
       // has changed and the CloudSdk#validate() code is out of date
       status = new Status(IStatus.WARNING, getClass().getName(),
-          Messages.getString("CloudSdkNotFound", sdk.getSdkPath())); //$NON-NLS-1$
+          Messages.getString("CloudSdkNotFound", location)); //$NON-NLS-1$
       return false;
-    } catch (AppEngineJavaComponentsNotInstalledException ex) {
+    } catch (AppEngineJavaComponentsNotInstalledException | InvalidJavaSdkException ex) {
```

Comment:
`InvalidJavaSdkException` is when it can't find the `bin/java` binary, so returning the `Messages.getString("AppEngineJavaComponentsNotInstalled", ...)` below seems a bit out of place.